cli/src/main/java/com/liferay/blade/cli/command/UpdateCommand.java

```diff
@@ -20,47 +20,430 @@
+		String updatePatch = matcher.group(3);
+
+		if (Integer.parseInt(updateMajor) == Integer.parseInt(currentMajor)) {
+			if (Integer.parseInt(updateMinor) == Integer.parseInt(currentMinor)) {
+				if (Integer.parseInt(updatePatch) == Integer.parseInt(currentPatch)) {
+					equal = true;
+				}
+			}
+		}
+
+		return equal;
+	}
+
+	public static String getUpdateJarUrl(boolean snapshots) throws IOException {
+		String url = RELEASES_REPO_URL;
+
+		if (snapshots) {
+			url = SNAPSHOTS_REPO_URL;
+		}
+
+		if (hasUpdateUrlFromBladeDir()) {
+			url = getUpdateUrlFromBladeDir();
+		}
+
+		String version = "";
+
+		String nextUrl = "";
+
+		String jarUrl = "";
+
+		Document versionsDocument;
+
+		Connection connection = Jsoup.connect(url);
+
+		versionsDocument = connection.get();
+
+		Elements tdOrPreElements = versionsDocument.select("td,pre");
+
+		for (Element potentialVersion : tdOrPreElements.select("a")) {
+			String bladeDir;
+
+			boolean prependUrl = false;
+
+			String href = potentialVersion.attr("href");
+
+			if (href.contains(url)) {
+				String hrefSubstring = href.substring(url .length());
+
+				bladeDir = hrefSubstring.replaceAll("/", "");
+			}
+			else {
+				bladeDir = href.replaceAll("/", "");
+
+				prependUrl = true;
+			}
+
+			if (bladeDir.matches("\\d+\\..*")) {
+				if (prependUrl) {
+					nextUrl = url + potentialVersion.attr("href");
+				}
+				else {
+					nextUrl = potentialVersion.attr("href");
+				}
+			}
+		}
+
+		if ("".equals(nextUrl)) {
+			throw new IOException("No directory found at url = " + url);
+		}
+
+		version = "nextUrl = " + nextUrl + "\n";
+
+		connection = Jsoup.connect(nextUrl);
+
+		Document jarsDocument = connection.get();
+
+		tdOrPreElements = jarsDocument.select("td,pre");
+
+		for (Element potentialJar : tdOrPreElements.select("a")) {
+			String bladeJar;
+
+			boolean prependUrl = false;
+
+			String href = potentialJar.attr("href");
+
+			if (href.contains(nextUrl)) {
+				String hrefSubstring = href.substring(nextUrl .length());
+
+				bladeJar = hrefSubstring.replaceAll("/", "");
+			}
+			else {
+				bladeJar = href.replaceAll("/", "");
+
+				prependUrl = true;
+			}
+
+			version = version + "\na jar = " + potentialJar.attr("href");
+
+			if (bladeJar.matches(".*\\.jar")) {
+				if (prependUrl) {
+					jarUrl = nextUrl + potentialJar.attr("href");
+				}
+				else {
+					jarUrl = potentialJar.attr("href");
+				}
+			}
+		}
+
+		if ("".equals(jarUrl)) {
+			throw new IOException("No jar found at nextUrl = " + nextUrl);
+		}
+
+		return jarUrl;
+	}
+
+	public static String getUpdateUrlFromBladeDir() {
+		String url = "no url";
+
+		final File updateUrlFile = new File(System.getProperty("user.home"), ".blade/update.url");
+
+		if (hasUpdateUrlFromBladeDir()) {
+			List<String> lines;
+
+			try {
+				lines = Files.readAllLines(Paths.get(updateUrlFile.getPath()), StandardCharsets.UTF_8);
+
+				url = lines.get(0);
+			}
+			catch (IOException ioe) {
+				ioe.printStackTrace();
```

Comment:
Can we use `getBladeCLI().error(ioe)`?

Reply:
Unfortunately this is a static command, so no access to getBladeCLI() from here.